### PR TITLE
Add a test plan for csproj-referencing-xproj scenarios

### PR DESCRIPTION
**Pull request by jasonmalinowski**
The test plan so far. Still to do:

- [ ] Add some steps for consuming an xproj from a Windows 8.1 Universal app.
- [ ] Add some steps for consuming an xproj from a Xamarin app.

*Review:* @davkean, @Pilchie, @rchande, @yishaigalatzer, @emgarten

### CODE REVIEW

File: docs/test-plans/csproj-referencing-xproj.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+1. Add a reference from the application to class library. *Expected:* the reference is correctly added.
+1. Go to the project reference in Solution Explorer, and click the project reference. Open the properties window, and verify the path mentions the artifact is coming from `$EXPECTEDTARGET`.
+1. Consume the class library inside the application by creating an instance of Class1 in Program.Main.
+1. At the use of the constructor, F12 on Class1. *Expected:* we correctly navigate to the source. We should **not** navigate to metadata.
+1. Set a breakpoint in the constructor inside the class library.
+1. F5 the application. *Expected:* the application runs, and the breakpoint is hit.
+
+### Testing Project Reference Closure
+
+1. Continuing with the desktop console app scenario, add another xproj project (ClassLibrary2) the same as the first one, and add Class2 to that project.
```
Comment (Pilchie) [inline docs/test-plans/csproj-referencing-xproj.md]: Make it explicit that the framework should line up.  Also, what about a test for a lower `netstandard` then the consuming side?

File: docs/test-plans/csproj-referencing-xproj.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    {
+      "dependencies": {
+        "NETStandard.Library": "1.0.0-rc2-23811"
+      },
+      "frameworks": {
+        "$PORTABLETARGET": { }
+      }
+    }
+    ```
+1. **Workaround:** If `$XPROJTARGET` starts with `netstandard`, insert `"imports": "imports": [ "portable-net45+win8", "dotnet5.6", "dnxcore50" ]`, where the number after dotnet should be the number you inserted for netstandard in the previous step, plus 4.1.
```
Comment (Pilchie) [inline docs/test-plans/csproj-referencing-xproj.md]: Stutter `"imports": "imports":`

File: docs/test-plans/csproj-referencing-xproj.md [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+1. Rebuild. *Expected:* the portable library builds.
+1. Add a reference from the portable class library. *Expected:* the reference is correctly added.
```
Comment (Pilchie) [inline docs/test-plans/csproj-referencing-xproj.md]: Add a reference to the xproj.

File: docs/test-plans/csproj-referencing-xproj.md [hunk 4]
```diff
@@ -0,0 +1,112 @@
...
+1. Consume the xproj project in the portable class library and F5 on a usage of a type from the xproj project. *Expected:* we correctly navigate to the source.
```
Comment (Pilchie) [inline docs/test-plans/csproj-referencing-xproj.md]: F12, not F5 here.

File: docs/test-plans/csproj-referencing-xproj.md [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+      },
+      "frameworks": {
+        ".NETPortable,Version=v4.5,Profile=Profile$PORTABLEPROFILENUMBER": { }
+      }
+    }
+    ```
+    making substitutions as necessary.
+1. Rebuild. *Expected:* the portable library builds.
+1. Add a reference from the portable class library. *Expected:* the reference is correctly added.
+1. Consume the xproj project in the portable class library and F5 on a usage of a type from the xproj project. *Expected:* we correctly navigate to the source.
```
Comment (Pilchie) [inline docs/test-plans/csproj-referencing-xproj.md]: F12, not F5.

**Comment by Pilchie** [comment]
I would add the todos directly to the document, even if you don't fill out the sub-bullet instructions for them yet.

**Comment by jasonmalinowski** [comment]
My intent is to merge with the TODOs done. I just wanted this out for review so far.

**Comment by jasonmalinowski** [comment]
Per the [updated roadmap](https://blogs.msdn.microsoft.com/dotnet/2016/05/06/net-core-rc2-improvements-schedule-and-roadmap/), this test plan probably doesn't need to exist in this form.

**Pull request closed without merging**